### DISCUSSION TITLE
How can structured logging be enabled for pushgateway?

### DISCUSSION THREAD

**Discussion by Ghazgkull** [prometheus/pushgateway#438] [Q&A]
## Feature request
Out of the box, pushgateway logs messages in flat text format. Are there flags that can be passed to enable structured json logging? This would improve pushgateway's observability by letting it better integrate with common log-shipping tools in use today.

If it's possible to do so, I'd be happy to capture the instructions in the README of this project and I would also volunteer to update the helm chart with log format options matching those used by the prometheus operator.

**Comment by beorn7** [comment]
Does `./pushgateway --log.format=json` help you?

**Comment by Ghazgkull** [comment]
Yes, actually this is exactly what I needed! The helm chart doesn't expose it directly, but you can always pass extra args like so:
```
extraArgs:
  - --log.format=json
  ```
